### PR TITLE
remove storage capability auto detection

### PR DESCRIPTION
**Pull request by dkeven**
Signed-off-by: dkven <dkvvven@gmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
The current auto-detection of StorageClasses' capabilities, by probing its CSIDriver's gRPC socket, does not work most of the time.

And when probe fails, it outputs a huge amount of error logs repeatedly, making things even worse.

To be backward compatible, and to be compatible with current ks-console logic, this PR removes the probing logic from the storageCapabilityController, and uses the preset ProvisionerCapabilities as the value source of StorageClassCapabilities instead.

That is, when a StorageClass/ProvisionerCapability Object gets created/updated, the controller will find the corresponding PC, and, based on its values, creates(if not exists)/updates the relevant StorageClassCapability Object(s), as well as updates the SCs' annotations or creates VolumeSnapshotClass Object(s), if necessary.

Therefore, with the new version of this controller, users need to mannually create ProvisionerCapability or StorageClassCapability Objects for their custom StorageClasses. (like they do now)

Also, this PR adds a preset ProvisionerCapability yaml file for the qingcloud-csi and qingstor-csi driver.

This is a minor, backward compatible, temporary fix. In the future, we will change ProvisionerCapability, and discard StorageClassCapability entirely, to provide a simple version of storage capability, which requires co-working with ks-console.

**Which issue(s) this PR fixes**:

Fixes (https://github.com/kubesphere/kubesphere/issues/3843)

### CODE REVIEW

File: config/default/provisonercapability/disk-csi-qingcloud-com.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: storage.kubesphere.io/v1alpha1
```
Comment (stoneshi-yunify) [inline config/default/provisonercapability/disk-csi-qingcloud-com.yaml:1]: I suspect this file and the below neonsan-csi-qingstor-com.yaml will be used by ks-installer to preinstall the provisonercapabilities, if so, be sure to submit related PR to ks-installer or talk with @pixiake

File: pkg/controller/storage/capability/capability_controller.go [hunk 2]
```diff
@@ -141,12 +126,20 @@ func NewController(
...
+	// but don't overdo by cascade deleting the SCCs when a PC is deleted
+	// since the role of PCs is more like a template rather than owner to SCCs
+
+	// This is a backward compatible fix to remove the useless auto detection of SCCs
+	// in the future, we will only keep ProvisionerCapability and remove the StorageClassCapability CRD entirely
+	ksStorageInformer.ProvisionerCapabilities().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: controller.handleProvisionerCapability,
 		UpdateFunc: func(oldObj, newObj interface{}) {
-			return
+			controller.handleProvisionerCapability(newObj)
```
Comment (stoneshi-yunify) [inline pkg/controller/storage/capability/capability_controller.go:140]: compare resourceversion of old and new object before this?

File: pkg/controller/storage/capability/capability_controller.go [hunk 3]
```diff
@@ -141,12 +126,20 @@ func NewController(
...
 		},
-		DeleteFunc: controller.handlerCSIDriver,
+		DeleteFunc: func(obj interface{}) {},
```
Comment (stoneshi-yunify) [inline pkg/controller/storage/capability/capability_controller.go:142]: can this DeleteFunc be leaved unset?

File: config/default/provisonercapability/disk-csi-qingcloud-com.yaml (hunk 1 above)
Comment (dkeven) [reply config/default/provisonercapability/disk-csi-qingcloud-com.yaml:1]: Yes, that's true, I'll do that when this PR is approved

File: pkg/controller/storage/capability/capability_controller.go [hunk 4]
```diff
@@ -278,14 +270,16 @@ func (c *StorageCapabilityController) syncHandler(key string) error {
 	if err != nil {
 		return err
 	}
-	// No capability because csi-plugin not installed
+	// Can't get StorageCapability's ProvisionerCapability
```
Comment (stoneshi-yunify) [inline pkg/controller/storage/capability/capability_controller.go:273]: Should be `StorageClass's ProvisionerCapability`?

**Review by stoneshi-yunify — CHANGES_REQUESTED** [review]
Review done. Main logic is okay to me, thanks. Though need to address my questions.

**Review by dkeven — COMMENTED** [review]
> Review done. Main logic is okay to me, thanks. Though need to address my questions.

Thanks for the review & suggestions, I've made the changes accordingly.

**Review by stoneshi-yunify — COMMENTED** [review]
/lgtm

**Pull request merged**